cpp/src/groupby/groupby.cu

```diff
@@ -71,19 +75,21 @@ groupby::~groupby() = default;
 
 namespace {
 /// Make an empty table with appropriate types for requested aggs
-auto empty_results(std::vector<aggregation_request> const& requests) {
+template <typename aggregation_request_t, typename F>
+auto templated_empty_results(std::vector<aggregation_request_t> const& requests, F get_kind) {
```

Comment:
I appreciate trying to reuse functionality, but this looks kinda hacky now.

Reply:
I posted a suggestion https://github.com/rapidsai/cudf/pull/3831#discussion_r368655801 that would also fix this https://github.com/rapidsai/cudf/pull/3831#discussion_r368653127

Reply:
I should have retained the function-name. I was trying to avoid the wholesale repetition of code there. The only difference was in how the `kind` of aggregation is fetched.
Revert?